### ISSUE TITLE
TinyMCE RC1 link_list

### ISSUE THREAD

**Issue by cfconsultancy** [tinymce/tinymce#4736]
I tried to add a link_list to tinyMCE RC1 and followed this instruction (looks like the same in tiny4) https://www.tiny.cloud/docs-beta/plugins/link/#link_list

The following error appears

`Error: Errors: 
Failed path: (dialog > body > branch: panel > items > [3] > branch: selectbox > items > [0])
Could not find valid *strict* value for "value" in {
  "text": "None",
  "value": null
}`

The link_list is a php file with a json output  and works in tinyMCE 4

`[{"title":"122-asdasdasdasdas2","value":"[LINK:20]"},{"title":"adres","value":"[LINK:7]"},{"title":"artiekel-teset","value":"[LINK:12]"},{"title":"artikelen","value":"[LINK:3]"},{"title":"asdasasdd","value":"[LINK:15]"},{"title":"asdasdasdas","value":"[LINK:14]"},{"title":"asdasdsd-asdasd-sdasdasdasdasdasdasdasdasd","value":"[LINK:19]"},{"title":"contact","value":"[LINK:2]"},{"title":"cvvsdffsdd-afgadfasdfasdf-asdfasdfasdf-adfasdfd-fsdfsdfsdfsdfdf","value":"[LINK:17]"},{"title":"dasdasdasasdasdasdas","value":"[LINK:16]"},{"title":"dfsdfsdfsdf","value":"[LINK:24]"},{"title":"facebook","value":"[LINK:13]"},{"title":"gallery","value":"[LINK:6]"},{"title":"googleplus","value":"[LINK:28]"},{"title":"index","value":"[LINK:1]"},{"title":"instagram","value":"[LINK:22]"},{"title":"nieuws","value":"[LINK:5]"},{"title":"producten","value":"[LINK:4]"},{"title":"sdfsdf-sdfsd-sdfsdf-sdfsdfd-zcvwertwasdfd-sdf","value":"[LINK:18]"},{"title":"test-3-en-4","value":"[LINK:9]"},{"title":"test1","value":"[LINK:23]"},{"title":"test2","value":"[LINK:8]"},{"title":"youtube","value":"[LINK:21]"},{"title":"zxsdfssdfsdf","value":"[LINK:10]"}]`

Even the example doesn't work
`link_list: function(success) {
    success([
       {title: 'My page 1', value: 'https://www.tiny.cloud'},
       {title: 'My page 2', value: 'https://about.tiny.cloud'}
    ]);
  },`

**Comment by lnewson** [comment]
Will have to test the XHR version as that requires some extra setup, but I was unable to replicate the example issue you mentioned. Here's a working example: http://fiddle.tinymce.com/H8gaab

So with that in mind, could you let us know what browser and OS you are using, as it be a browser specific issue?

**Comment by lnewson** [comment]
Tried to replicate the issue with a PHP and JSON file as well, however I was unable to replicate it, as the link dialog correctly contained the items listed in the external file.

If it helps, here's the exact content of the PHP file I used for testing:
```
<?php
  header('Content-Type: application/json');
  echo '[{"title":"122-asdasdasdasdas2","value":"[LINK:20]"},{"title":"adres","value":"[LINK:7]"},{"title":"artiekel-teset","value":"[LINK:12]"},{"title":"artikelen","value":"[LINK:3]"},{"title":"asdasasdd","value":"[LINK:15]"},{"title":"asdasdasdas","value":"[LINK:14]"},{"title":"asdasdsd-asdasd-sdasdasdasdasdasdasdasdasd","value":"[LINK:19]"},{"title":"contact","value":"[LINK:2]"},{"title":"cvvsdffsdd-afgadfasdfasdf-asdfasdfasdf-adfasdfd-fsdfsdfsdfsdfdf","value":"[LINK:17]"},{"title":"dasdasdasasdasdasdas","value":"[LINK:16]"},{"title":"dfsdfsdfsdf","value":"[LINK:24]"},{"title":"facebook","value":"[LINK:13]"},{"title":"gallery","value":"[LINK:6]"},{"title":"googleplus","value":"[LINK:28]"},{"title":"index","value":"[LINK:1]"},{"title":"instagram","value":"[LINK:22]"},{"title":"nieuws","value":"[LINK:5]"},{"title":"producten","value":"[LINK:4]"},{"title":"sdfsdf-sdfsd-sdfsdf-sdfsdfd-zcvwertwasdfd-sdf","value":"[LINK:18]"},{"title":"test-3-en-4","value":"[LINK:9]"},{"title":"test1","value":"[LINK:23]"},{"title":"test2","value":"[LINK:8]"},{"title":"youtube","value":"[LINK:21]"},{"title":"zxsdfssdfsdf","value":"[LINK:10]"}]';
?>
```

**Comment by cfconsultancy** [comment]
It looks like the rel_list is this problem, if the value is empty the error occurs

http://fiddle.tinymce.com/H8gaab/3

 {title: 'None', value: ''}, error
 {title: 'None', value: 'anything'}, works

  rel_list: [
    {title: 'None', value: ''},
    {title: 'Lightbox', value: 'lightbox'},
    {title: 'Table of contents', value: 'toc'}
  ],

**Comment by cfconsultancy** [comment]
Also the link_list creates the following Ui code

<select size="1" data-alloy-tabstop="true" tabindex="-1" id="form-field_2537790613431547038364792">

Is it possible to set a class name on that selectbox ? So we can use Chosen jquery ( https://harvesthq.github.io/chosen/ )

**Comment by lnewson** [comment]
This should be fixed in the RC2 release of 5.0. As for the other question about class names, that's been answered here: https://github.com/tinymce/tinymce/issues/4761
